src/jit/codegenxarch.cpp

```diff
@@ -2706,6 +2726,7 @@ CodeGen::genStoreRegisterReturnInLclVar(GenTreePtr treeNode)
         if (type0 != TYP_UNKNOWN)
         {
             if (structDesc.eightByteClassifications[0] == SystemVClassificationTypeIntegerReference ||
+                structDesc.eightByteClassifications[0] == SystemVClassificationTypeIntegerByReference ||
```

Comment:
Using the new IsIntegral/SseSlot methods. The code for type0 and type1 is doing different things - one is setting the firstInt/FloatUsed flags and the other is testing them.